src/memory_backing/mod.rs

```diff
@@ -0,0 +1,24 @@
+// pub mod byte_field;
+mod vec;
+
+pub trait MemoryBacking<T> {
+    fn new() -> Self
+    where
+        Self: Sized;
+
+    fn with_capacity(size: usize) -> Self
+    where
+        Self: Sized;
+
+    fn len(&self) -> usize;
+
+    fn push(&mut self, item: T);
+
+    fn index(&self, index: usize) -> &T;
+
+    fn index_mut(&mut self, index: usize) -> &mut T;
```

Comment:
By analogy with `Vec` and other data structures, get returns an `Option<..>` that fails gracefully if out of bounds, etc. The index just assumes it will be in bounds and throws an error otherwise, matching these methods

Reply:
Fair enough 👌